src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -1165,16 +1166,16 @@ DecideTransactionPropertiesForTaskList(RowModifyLevel modLevel, List *taskList,
 		return xactProperties;
 	}
 
-	if (GetCurrentLocalExecutionStatus() == LOCAL_EXECUTION_REQUIRED)
+	if (GetCurrentLocalExecutionStatus() == LOCAL_EXECUTION_PERFORMED_MODIFICATION)
 	{
 		/*
-		 * In case localExecutionHappened, we force the executor to use 2PC.
-		 * The primary motivation is that at this point we're definitely expanding
-		 * the nodes participated in the transaction. And, by re-generating the
-		 * remote task lists during local query execution, we might prevent the adaptive
-		 * executor to kick-in 2PC (or even start coordinated transaction, that's why
-		 * we prefer adding this check here instead of
-		 * Activate2PCIfModifyingTransactionExpandsToNewNode()).
+		 * In case a local execution with modification happened, we force the
+		 * executor to use 2PC. The primary motivation is that at this point
```

Comment:
do we also need to do this if this command is not a modification?

Reply:
(if not, we should probably change the name to *Should*Use2PC)

Reply:
With SELECT FOR UPDATE, we do not use 2PC. As far as I can say, that is expected behaviour, right?

Reply:
Seems fine to me..